particle-iot/device-os
# chore/sc-101315/test-build-system-tune-timeouts

### Problem

We've continue to see sporadic failures over in the `test-build-system` CI workflow related to `apt-get install` connection timeouts ([example](https://app.circleci.com/pipelines/github/particle-iot/device-os/712/workflows/f301cf74-1c81-4335-bc52-891fd8547e2c/jobs/11952?invite=true#step-103-122))

### Solution

Increase `apt` timeout and retry attempts, increase CircleCI's `no_output_timeout` option to allow more time for build tasks to complete

### Steps to Test

1. Review the [updated CI/CD pipeline](https://github.com/particle-iot/device-os/blob/chore/sc-101315/test-build-system-tune-timeouts/.circleci/config.yml#L298-L307)
5. Observe the [latest CI run in CircleCI](https://app.circleci.com/pipelines/github/particle-iot/device-os/715/workflows/f275bb8e-a257-46fb-a4d1-ed180fa4be8d)

**outcome**

CI workflow should pass, changes should be appropriate

### References

https://app.shortcut.com/particle/story/101315/device-os-test-build-system-ci-workflow-occasionally-errors
https://app.circleci.com/pipelines/github/particle-iot/device-os?filter=all
https://github.com/particle-iot/device-os/pull/2444
https://github.com/particle-iot/device-os/pull/2434
https://github.com/particle-iot/device-os/pull/2418


---

### Completeness

- [] Run unit/integration/application tests on device
